src/internal/converters/eml/eml.go

```diff
@@ -25,75 +26,105 @@ const (
+	name := ptr.Val(entry.GetName())
+	email := ptr.Val(entry.GetAddress())
+
+	if name == email || len(name) == 0 {
+		return email
+	}
+
+	return fmt.Sprintf(addressFormat, name, email)
+}
+
 // FromJSON converts a Messageable (as json) to .eml format
 func FromJSON(ctx context.Context, body []byte) (string, error) {
 	data, err := api.BytesToMessageable(body)
 	if err != nil {
 		return "", clues.Wrap(err, "converting to messageble")
 	}
 
+	ctx = clues.Add(ctx, "id", ptr.Val(data.GetId()))
+
 	email := mail.NewMSG()
 
 	if data.GetFrom() != nil {
-		email.SetFrom(
-			fmt.Sprintf(
-				addressFormat,
-				ptr.Val(data.GetFrom().GetEmailAddress().GetName()),
-				ptr.Val(data.GetFrom().GetEmailAddress().GetAddress())))
+		email.SetFrom(formatAddress(data.GetFrom().GetEmailAddress()))
+
+		if email.Error != nil {
+			return "", clues.Wrap(email.Error, "adding from address").
+				WithClues(ctx).
+				With("from", data.GetFrom())
+		}
 	}
 
 	if data.GetToRecipients() != nil {
 		for _, recipient := range data.GetToRecipients() {
-			email.AddTo(
-				fmt.Sprintf(
-					addressFormat,
-					ptr.Val(recipient.GetEmailAddress().GetName()),
-					ptr.Val(recipient.GetEmailAddress().GetAddress())))
+			email.AddTo(formatAddress(recipient.GetEmailAddress()))
+
+			if email.Error != nil {
+				return "", clues.Wrap(email.Error, "adding to address").
+					WithClues(ctx).
+					With("to", recipient)
+			}
 		}
 	}
 
 	if data.GetCcRecipients() != nil {
 		for _, recipient := range data.GetCcRecipients() {
-			email.AddCc(
-				fmt.Sprintf(
-					addressFormat,
-					ptr.Val(recipient.GetEmailAddress().GetName()),
-					ptr.Val(recipient.GetEmailAddress().GetAddress())))
+			email.AddCc(formatAddress(recipient.GetEmailAddress()))
+
+			if email.Error != nil {
+				return "", clues.Wrap(email.Error, "adding cc address").
+					WithClues(ctx).
+					With("cc", recipient)
+			}
 		}
 	}
 
 	if data.GetBccRecipients() != nil {
 		for _, recipient := range data.GetBccRecipients() {
-			email.AddBcc(
-				fmt.Sprintf(
-					addressFormat,
-					ptr.Val(recipient.GetEmailAddress().GetName()),
-					ptr.Val(recipient.GetEmailAddress().GetAddress())))
+			email.AddBcc(formatAddress(recipient.GetEmailAddress()))
+
+			if email.Error != nil {
+				return "", clues.Wrap(email.Error, "adding bcc address").
+					WithClues(ctx).
+					With("bcc", recipient)
+			}
 		}
 	}
 
 	if data.GetReplyTo() != nil {
 		rts := data.GetReplyTo()
 		if len(rts) > 1 {
 			logger.Ctx(ctx).
-				With("id", ptr.Val(data.GetId()),
-					"reply_to_count", len(rts)).
-				Warn("more than 1 reply to")
-		} else if len(rts) != 0 {
-			email.SetReplyTo(
-				fmt.Sprintf(
-					addressFormat,
-					ptr.Val(rts[0].GetEmailAddress().GetName()),
-					ptr.Val(rts[0].GetEmailAddress().GetAddress())))
+				With("reply_to_count", len(rts)).
+				Warn("more than 1 Reply-To, adding only the first one")
+		}
+
+		if len(rts) != 0 {
+			email.SetReplyTo(formatAddress(rts[0].GetEmailAddress()))
```

Comment:
do we need to check for `email.Error` here as well? The other assignments seem to do that

Reply:
oh, actually it seems there's a `GetError()` [function](https://pkg.go.dev/github.com/xhit/go-simple-mail/v2#Email.GetError). With that it may be possible to just add all the data and then check for errors once at the end (assuming the error message returned by the go-simple-email package is descriptive as to what failed

Reply:
Yeah, it looks like the errors are descriptive enough. Let me just switch to a single error.